### PR TITLE
added appium native tests for javascript selenium 3 and 4

### PR DESCRIPTION
**Pull request by Artem-applitools**
Here new implementation of the appium native tests on android and iOS
Selenium 4 have an issues with baselines for native apps sets browser name to unknown. Also iOS tests extremely slow on S4 SDK.

### CODE REVIEW

**Comment by amitzur** [comment]
@Artem-applitools I'm getting this exception when running `npx mocha --no-timeouts packages/eyes-selenium/test/coverage/custom/TestAppiumNative.spec.js`:
```
  TestAppiumNative
(node:28518) UnhandledPromiseRejectionWarning: InvalidSelectorError: Locator Strategy 'css selector' is not supported for this session
    at Object.checkLegacyResponse (/home/amit/applitools/eyes.sdk.javascript1/node_modules/selenium-webdriver/lib/error.js:585:15)
    at parseHttpResponse (/home/amit/applitools/eyes.sdk.javascript1/node_modules/selenium-webdriver/lib/http.js:556:13)
    at Executor.execute (/home/amit/applitools/eyes.sdk.javascript1/node_modules/selenium-webdriver/lib/http.js:491:26)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async WebDriver.execute (/home/amit/applitools/eyes.sdk.javascript1/node_modules/selenium-webdriver/lib/webdriver.js:700:17)
(node:28518) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
(node:28518) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

File: packages/eyes-selenium-3/test/coverage/custom/TestAppiumNative.spec.js [hunk 1]
```diff
@@ -30,9 +30,93 @@ describe('TestAppiumNative', () => {
...
+        .usingServer(sauceUrl)
+        .build()
+      eyes = new Eyes()
+      eyes.setBatch(batch)
+      eyes.setBranchName('master')
+    })
+
+    it('AndroidNativeApp checkWindow', async () => {
+      await eyes.open(driver, 'AndroidNativeApp', 'AndroidNativeApp checkWindow')
+      await eyes.check('', Target.window().ignore({left: 1271, top: 0, width: 158, height: 100}))
```
Comment (amitzur) [inline packages/eyes-selenium-3/test/coverage/custom/TestAppiumNative.spec.js:46]: there's no api assertion of the ignore region

File: packages/eyes-selenium-3/test/coverage/custom/TestAppiumNative.spec.js [hunk 2]
```diff
@@ -30,9 +30,93 @@ describe('TestAppiumNative', () => {
...
+          maxRightOffset: 3,
+          maxUpOffset: 20,
+          maxDownOffset: 30,
+        }),
+      )
+      await eyes.close()
+    })
+  })
+
   it(`Native app on sauce lab`, async () => {
```
Comment (amitzur) [inline packages/eyes-selenium-3/test/coverage/custom/TestAppiumNative.spec.js:117]: what's the point of this test, given the android tests above?

File: packages/eyes-selenium/test/coverage/custom/TestAppiumNative.spec.js [hunk 3]
```diff
@@ -48,49 +120,27 @@ describe('TestAppiumNative', () => {
     await eyes.close()
   })
 
-  it.skip(`Native iOS app on sauce lab`, async () => {
```
Comment (amitzur) [inline packages/eyes-selenium/test/coverage/custom/TestAppiumNative.spec.js:51]: what's the point of this test, given the iOS tests above?

File: packages/eyes-selenium/test/coverage/custom/TestAppiumNative.spec.js [hunk 4]
```diff
@@ -31,9 +24,88 @@ describe('TestAppiumNative', () => {
...
+          3,
+          20,
+          30,
+        ),
+      )
+      await eyes.close()
+    })
+  })
+
   it(`Native android app on sauce lab`, async () => {
```
Comment (amitzur) [inline packages/eyes-selenium/test/coverage/custom/TestAppiumNative.spec.js:106]: what's the point of this test, given the android tests above?

File: packages/eyes-selenium-3/test/coverage/custom/TestAppiumNative.spec.js (hunk 1 above)
Comment (Artem-applitools) [reply packages/eyes-selenium-3/test/coverage/custom/TestAppiumNative.spec.js:46]: Ignore region was added cause app used in the tests have a status bar with a clock, so they display different time from run to run and makes differences so ignore region was added.

**Comment by Artem-applitools** [comment]
@amitzur I will clear the redundant tests. 
The exception don't fail the tests. It displayed only warning, not an exception. Test is actually passes, but it takes some more time after the warning.

**Comment by amitzur** [comment]
@Artem-applitools what's the status of this PR? Is it included in #188 ? How should we move forward here?

**Comment by Artem-applitools** [comment]
@amitzur I haven't added it to the [#188](url). Currently another QA working on implementing mobile tests as a generic tests. I think it would be better to add them as generic when they will be implemented. So I will close this PR.

**Comment by amitzur** [comment]
@Artem-applitools sounds amazing. Thanks!

**Pull request closed without merging**